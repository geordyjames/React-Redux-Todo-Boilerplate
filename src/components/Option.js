import React from 'react';

const Option = props => (
	<li>
		<p className="uk-text-middle uk-align-left uk-margin-small-bottom">
			{props.count}. {props.optionText}
		</p>
		<button
			className="uk-icon-button uk-align-right"
			data-uk-icon="icon: trash"
			onClick={() => {
				props.handleDeleteOption(props.id);
			}}
		/>
	</li>
);

export default Option;
