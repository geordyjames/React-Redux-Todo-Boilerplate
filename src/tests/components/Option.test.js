import React from 'react';
import { shallow } from 'enzyme';
import Option from '../../components/Option';

test('should render Option correctly', () => {
  const wrapper = shallow(<Option />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Option with values correctly', () => {
  const wrapper = shallow(<Option id={10} count={5} optionText="Hello" />);
  expect(wrapper).toMatchSnapshot();
});

test('should call handleDeleteOption on button click', () => {
  const handleDeleteOption = jest.fn();
  const wrapper = shallow(<Option id={10} handleDeleteOption={handleDeleteOption} />);
  wrapper.find('button').simulate('click');
  expect(handleDeleteOption).toHaveBeenLastCalledWith(10);
});