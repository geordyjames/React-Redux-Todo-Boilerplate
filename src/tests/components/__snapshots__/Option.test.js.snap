// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Option correctly 1`] = `
<li>
  <p
    className="uk-text-middle uk-align-left uk-margin-small-bottom"
  >
    . 
  </p>
  <button
    className="uk-icon-button uk-align-right"
    data-uk-icon="icon: trash"
    onClick={[Function]}
  />
</li>
`;

exports[`should render Option with values correctly 1`] = `
<li>
  <p
    className="uk-text-middle uk-align-left uk-margin-small-bottom"
  >
    5
    . 
    Hello
  </p>
  <button
    className="uk-icon-button uk-align-right"
    data-uk-icon="icon: trash"
    onClick={[Function]}
  />
</li>
`;
